### PR TITLE
Removing gnacs dependency

### PR DESCRIPTION
**Pull request by fionapigott**
I've removed all gnacs dependencies.

- Update setup.py to reflect the new dependency (tweet_parser)
- Edit the code to use tweet_parser properties anywhere it accesses a value of a tweet
- compatibility with Python 2 and Python 3
- tested by adding my API creds to test_search.sh and running bash test_search.sh
- updated the version number to 1.2.0 (last version on pypi is 1.1.0)

### CODE REVIEW

**Comment by fionapigott** [comment]
Also, once this is merged we should get added to the "gapi" pypi package so that we can update that.

File: gnip_search.py [hunk 1]
```diff
@@ -74,8 +78,8 @@ def __init__(self, token_list_size=40):
             sys.exit(-1)
 
         # Gnacs is not yet upgraded to python3, so don't allow CSV output option (which uses Gnacs) if python3
```
Comment (jeffakolb) [inline gnip_search.py]: Do we want references to Gnacs in here?

Comment (fionapigott) [inline gnip_search.py]: good call, I changed that comment

**Comment by fionapigott** [comment]
I removed references to search V1 vs V2, and changed the URL validation to allow data-api urls as well as gnip-api urls (however, the "gnip-api" urls are the official url, so I only suggest that domain in the error message)

**Pull request merged**